### PR TITLE
Wildcard V2: `<FeedbackPrompt />` Manual migration

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
### Refs
[SourceGraph Issue](https://github.com/sourcegraph/sourcegraph/issues/27713)
[GitStart ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-27713)
## Implementation
1. Find all usage of [FeedbackPrompt](https://sourcegraph.com/search?q=context:global+repo:github.com/sourcegraph/sourcegraph+%3CFeedbackPrompt+OR+FeedbackPromptContent&patternType=literal).
2. Notice how some components use FeedbackPromptContent directly. This isn't ideal, we should aim to only support usage of `FeedbackPrompt.`.
3. The FeedbackPrompt component wraps the prompt in a ButtonDropdown. We should make it so this component is just a prompt, and the consumer has to provide their own button. We should change this component slightly, look at [this solution](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@tr/bootstrap-transition/-/blob/client/web/src/enterprise/insights/pages/dashboards/dashboard-page/DashboardsPage.tsx#L94:13) and [this solution](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@tr/bootstrap-transition/-/blob/client/web/src/extensions/extension/SourcegraphExtensionFeedback.tsx#L26:16) as possible options.
4. Remove `FeedbackPromptContent` as an export. Update all usage to `<FeedbackPrompt />`.

### Acceptance criteria
- [ ] All incorrect usage of the previous pattern has been updated to use the Wildcard component

### CODE REVIEW

File: internal/search/job/job.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ type Args struct {
 // query on all indexed repositories) then we need to convert our tree to
```
Comment (umpox) [inline internal/search/job/job.go:46]: Please revert changes to this file!

Comment (phunhakent) [reply internal/search/job/job.go:46]: @umpox Thanks for taking a look, funny that this file on main branch has the same content as its in this branch, not sure why github show these changes here [sourcegraph search](https://sourcegraph.com/search?q=context:global+func+ToSearchJob%28jargs+*Args%2C+q+query.Q%29+%28run.Job%2C+error%29+%7B&patternType=literal)

Could you please review other parts? I'm finding a way to resolve this github issue

Comment (gitstart-sourcegraph) [reply internal/search/job/job.go:46]: @umpox Thanks for taking a look, funny that this file on main branch has the same content as its in this branch, not sure why github show these changes here [sourcegraph search](https://sourcegraph.com/search?q=context:global+func+ToSearchJob%28jargs+*Args%2C+q+query.Q%29+%28run.Job%2C+error%29+%7B&patternType=literal)

Could you please review other parts? I'm finding a way to resolve this github issue

File: client/wildcard/src/components/Feedback/FeedbackPrompt/FeedbackPrompt.tsx [hunk 2]
```diff
@@ -149,61 +143,95 @@ export const FeedbackPromptContent: React.FunctionComponent<ContentProps> = ({
...
+    }, [onClose])
+
+    const toggleIsOpen = useCallback(() => {
+        setIsOpen(isOpen => !isOpen)
+    }, [])
+
+    const triggerElement = ChildrenComponent ? <ChildrenComponent isOpen={isOpen} onClick={toggleIsOpen} /> : children
+
+    const contentElement = (
+        <FeedbackPromptContent onSubmit={onSubmit} productResearchEnabled={true} onClose={handleClosePrompt} />
```
Comment (umpox) [inline client/wildcard/src/components/Feedback/FeedbackPrompt/FeedbackPrompt.tsx:213]:
```suggestion
    modalLabelId = 'sourcegraph-feedback-modal',
    productResearchEnabled,
}) => {
    const [isOpen, setIsOpen] = useState(() => !!openByDefault)
    const ChildrenComponent = typeof children === 'function' && children
    const handleClosePrompt = useCallback(() => {
        setIsOpen(false)
        onClose?.()
    }, [onClose])

    const toggleIsOpen = useCallback(() => {
        setIsOpen(isOpen => !isOpen)
    }, [])

    const triggerElement = ChildrenComponent ? <ChildrenComponent isOpen={isOpen} onClick={toggleIsOpen} /> : children

    const contentElement = (
        <FeedbackPromptContent onSubmit={onSubmit} productResearchEnabled={productResearchEnabled} onClose={handleClosePrompt} />
```

We only want to show the product research message for the prompt in the nav bar

File: client/web/src/nav/GlobalNavbar.tsx [hunk 3]
```diff
@@ -269,7 +282,18 @@ export const GlobalNavbar: React.FunctionComponent<Props> = ({
                     )}
                     {props.authenticatedUser && (
                         <NavAction>
-                            <FeedbackPrompt routes={props.routes} />
+                            <FeedbackPrompt onSubmit={handleSubmitFeedback}>
```
Comment (umpox) [inline client/web/src/nav/GlobalNavbar.tsx:285]:
```suggestion
                            <FeedbackPrompt onSubmit={handleSubmitFeedback} productResearchEnabled={true}>
```

**Review by umpox — COMMENTED** [review]
Looking good, just noticed this

**Review by umpox — APPROVED** [review]
One final change to make and then this LGTM.

**Review by umpox — APPROVED** [review]
Thanks!

**Pull request merged**